WIP 0.5.0

Keats opened the pull request.

Commit:
Start work on 0.4.3

Commit:
Update deps and fix deprecrations

Commit:
Fix loading html files in themes outside of templates

Fix #412

tshepang commented on CHANGELOG.md:
```diff
@@ -1,5 +1,10 @@
 
+## 0.4.3 (unreleased)
+
+- Gutenberg has changed name to REPLACE_ME!
```
ah, interesting!

Keats replied:
See https://github.com/Keats/gutenberg/issues/377 for the discussion.
The new name will be `zola`

tshepang replied:
hehe, interesting... we got a famous celebrity (musician, actor, presenter) in South Africa using that as a nick

porglezomp committed:
Change highlighting to not include duplicated background colors

Keats committed:
Merge pull request #416 from porglezomp-misc/background-color

Change highlighting to not include duplicated background colors

Commit:
Mention highlight bg improvement in changelog

Commit:
Fix email links being checked by link checker

Closes #403

Commit:
Enable markdown extensions for markdown filter

Closes #417

Commit:
Default get_url to not add a trailing slash

Closes #388

Commit:
Fix base-url not overriding processed images urls

Closes #418

Commit:
Update deps

Commit:
Fix ignored shortcode swallowing whitespace

Fix #383

Commit:
Move to non-deprecated fns

Freaky committed:
Replace fold/reduce over Result::and with collect

Not only is this tidier, but it avoids making these iterators
Rayon-specific.

Freaky committed:
Use a HashSet for detecting orphan pages

This offers a big performance bump for large sites:

  small-kb (100 pages)
    before: 530ms
     after: 500ms

  medium-kb (1000 pages)
    before: 2.9s
     after: 1.5s

  huge-kb (10,000 pages)
    before: 150s
     after:   9.1s

Freaky committed:
Make get_all_orphan_pages more idiomatic

Keats committed:
Merge pull request #424 from Freaky/hash-orphans

Use a HashSet for detecting orphan pages

Commit:
Ignore one more type of emacs temp file

Commit:
Do not load markdown files starting with a .

Keats committed:
Merge pull request #427 from Freaky/foldreduce-to-collect

Replace fold/reduce over Result::and with collect

Commit:
_processed_images -> processed_images

Commit:
No replacein markdown rendering

Fix #430

Commit:
Some more info on installation

Commit:
Remove useless benches and add more useful ones

Commit:
Some more benches

Commit:
Avoid the use of enumerate in render_paginated

Commit:
Do not copy pages of sections when rendering taxonomies

Commit:
Only compute reading analytics once...

Commit:
Fix incorrect default for highlight_code of Config

Commit:
No clone when rendering rss feed

Commit:
Do not clone pages in taxonomies

Commit:
Parse date only once for pages

greizgh committed:
Allow to disable colors through env vars

- unless forced, only use colors in tty
- support CLICOLOR and NO_COLOR vars
- switch to termcolor crate to handle output coloring

Keats committed:
Merge pull request #438 from greizgh/feat-no-color

Allow to disable colors

Commit:
Mention a paginated section doesnt get its pages

Commit:
Update changelog

Commit:
Fix robots.txt not being loaded in Tera

Closes #443

Commit:
Update reqwest and image

Commit:
Correct usage of reqwest headers

ccoors committed:
Fix some clippy warnings

Commit:
Remove printlnt

Commit:
Try getting CI to work on next branch

Commit:
Remove paginator.pagers

Keats committed:
Merge pull request #455 from ccoors/fix-clippy

Fix some clippy warnings

Commit:
Fix robots.txt template for real this time

greizgh committed:
Inject config in robots.txt context

This allow to use base_url in robots.txt, to reference a sitemap for
example.

Keats committed:
Merge pull request #458 from greizgh/robots-config

Feat: inject config in robots.txt context

vsupalov committed:
Replace hard-coded directory slash in serve output (#457)

* Replace hard-coded directory slash in serve output

Commit:
Slotmap refactor

Commit:
Use Tera render_with_borrowed

Commit:
Cleanup of slotmap impl

Fix #205

Commit:
Rebuild more things now that it is more performant to do so

Fix #122

Commit:
Bump minor version to 1.28

Commit:
Remove debug symbols on release

Keats committed:
Merge pull request #459 from Keats/database

Slotmap refactor

Commit:
Update to syntect 3

Commit:
Update pest for shortcode parser

Commit:
Cargo update

jwatt committed:
Fix content parser's 'text' rule to use the correct production

chriskrycho committed:
Allow inclusion of all items in RSS feeds.

Switch to an `Option<usize>` for the serialized value of `rss_items`.
This lets us just set a blank value in the configuration and thereby
include *all* items.

This is a backwards-compatible change; it does not affect the behavior
of existing configurations.

Fixes #468. Closes #471.

Keats committed:
Merge pull request #472 from chriskrycho/make-rss_limit-optional

Allow inclusion of all items in RSS feeds.

Commit:
Clearer error message when failing to load theme.toml

Commit:
Remove line trimming for shortcode bodies

Can't remember why it was doing that even
with the comment and the test added in the
commit still pass so...

Fix #462

Commit:
Speed up travis

Keats committed:
Merge pull request #474 from jwatt/next

Fix content parser's 'text' rule to use the correct production

Commit:
templates folder is now optional for serve

Fix #445

Commit:
Add Reason syntax highlighting

Commit:
Clarify where to use shortcodes

chriskrycho committed:
Add F# syntax.

Keats committed:
Merge pull request #483 from chriskrycho/fsharp-syntax

Add F# syntax highlighting.

Commit:
Bundle the F# syntax

Commit:
section.subsections is now an array of paths

Close #446
Close #260
Close #478
Close #284
Close #480

bdjnk committed:
prevent html tags from appearing in the toc

Commit:
Have a list of ancestors instead of only parent section

Keats committed:
Merge pull request #488 from bdjnk/remove-tags-from-toc

prevent html tags from appearing in the toc

Commit:
Add metadata_only to get_section

Keats committed:
Merge pull request #484 from Keats/subsections

section.subsections is now an array of paths

Commit:
Mention toc changes in changelog

kellpossible committed:
CSV and TOML loading global functions (#379)

Local CSV/TOML/JSON loading Tera function

Commit:
Update changelog

Commit:
Expose relative path of pages & sections

Closes #485

Commit:
Do not have trailing slash for rss.xml

Closes #486

Commit:
Rename all occurrences of gutenberg to zola in code

Commit:
Update docs to refer to zola

Commit:
No need for clone

jwatt committed:
Remove unmatchable rules from render_shortcodes

Commit:
Always populate sections on section change

Commit:
Clarify redirect_to in docs

Fix #490

Keats committed:
Merge pull request #489 from jwatt/next

Remove unmatchable rules from render_shortcodes

Commit:
Do not panic if something is already bound to 1111 in serve

maghoff committed:
Ensure root output directory exists explicitly.

Aliases that have no directory nesting sneakily avoid the code path that ensures the directories exist

Keats committed:
Merge pull request #496 from maghoff/next

Ensure root output directory exists explicitly.

Commit:
Fix test

Commit:
Do not paginate drafts

Fix #495

Commit:
Small refactor for serialized page/sections

Commit:
Serialize page/section assets only once

Commit:
Set date from filename

Fix #396

Commit:
Add ayu highlighting theme

Commit:
Do not error on files starting with utf-8 BOM

Close #501

Commit:
Add manual download to netlify docs

RealOrangeOne committed:
Remote data (#494)

Commit:
Cargo update & some doc tweak

Commit:
Some clippy fixes

Commit:
rustfmt

Keats committed:
Merge branch 'master' into next

Commit:
Update Tera

Commit:
Fix load_toml date handling and fix bug in date conversion

Commit:
add --watch-only flag

Commit:
add --watch-only to docs

Commit:
remove debug-println!

Commit:
reset color in colorize

Keats committed:
Merge pull request #509 from FreeMasen/fix-color

reset color in colorize

Commit:
Fix build for 1.28

Keats committed:
Merge pull request #450 from FreeMasen/next

add watch command

Commit:
Update changelog

Commit:
Fix 1.28 build for real

Commit:
Add transparent sections

Closes #413
Closes #408

Commit:
Add page_template to sections

To override all child pages

Closes #397
Tests from PR #434

Keats committed:
Merge branch 'master' into next

Commit:
Fewer string cloning for tpl names

Commit:
Do not clone tpl name for pages

Commit:
Add test for nested page_template

Commit:
cargo update

Commit:
Handle file/dir renaming

Closes #385

Commit:
Correct changelog

Commit:
Update syntaxes

gsomix committed:
Add info about chocolatey package

Keats committed:
Merge pull request #517 from gsomix/patch-1

Add info about chocolatey package

Commit:
Update docs for install/netlify deploy for zola/gutenberg

Commit:
Rebuild completions

Commit:
Mention another breaking change design wise

Commit:
Fix bug with pages starting with dates

Commit:
Update deps + fmt

Commit:
Point to https submodule instead of git

Commit:
Ignore rename of non md files

Commit:
Fix shortcodes <-> markdown indentation

Commit:
Fix reload with taxonomies

Commit:
Min Rust version updated to 1.29

Commit:
Update tokens and badges

Commit:
Set release date
